cocos/renderer/CCGLProgram.cpp

```diff
@@ -236,50 +249,55 @@ bool GLProgram::initWithByteArrays(const GLchar* vShaderByteArray, const GLchar*
 
 bool GLProgram::initWithByteArrays(const GLchar* vShaderByteArray, const GLchar* fShaderByteArray, const std::string& compileTimeDefines)
 {
+    return initWithByteArrays(vShaderByteArray, fShaderByteArray, "", compileTimeDefines);
+}
+
+bool GLProgram::initWithByteArrays(const GLchar* vShaderByteArray, const GLchar* fShaderByteArray, const std::string& compileTimeHeaders, const std::string& compileTimeDefines)
+{
     _program = glCreateProgram();
     CHECK_GL_ERROR_DEBUG();
-
+    
     // convert defines here. If we do it in "compileShader" we will do it twice.
     // a cache for the defines could be useful, but seems like overkill at this point
     std::string replacedDefines = "";
     replaceDefines(compileTimeDefines, replacedDefines);
-
+    
     _vertShader = _fragShader = 0;
-
+    
     if (vShaderByteArray)
     {
-        if (!compileShader(&_vertShader, GL_VERTEX_SHADER, vShaderByteArray, replacedDefines))
+        if (!compileShader(&_vertShader, GL_VERTEX_SHADER, vShaderByteArray, compileTimeHeaders, replacedDefines))
         {
             CCLOG("cocos2d: ERROR: Failed to compile vertex shader");
             return false;
-       }
+        }
     }
-
+    
     // Create and compile fragment shader
     if (fShaderByteArray)
     {
-        if (!compileShader(&_fragShader, GL_FRAGMENT_SHADER, fShaderByteArray, replacedDefines))
+        if (!compileShader(&_fragShader, GL_FRAGMENT_SHADER, fShaderByteArray, compileTimeHeaders, replacedDefines))
         {
             CCLOG("cocos2d: ERROR: Failed to compile fragment shader");
             return false;
         }
     }
-
+    
     if (_vertShader)
     {
         glAttachShader(_program, _vertShader);
     }
     CHECK_GL_ERROR_DEBUG();
-
+    
     if (_fragShader)
     {
         glAttachShader(_program, _fragShader);
     }
-
+    
```

Comment:
pease fix indent issue, remove unneeded spaces, there are many places have this issue